apps/cpp_rpc/rpc_env.cc

```diff
@@ -86,7 +86,13 @@ void CleanDir(const std::string& dirname);
 std::string BuildSharedLibrary(std::string file_in);
 
 RPCEnv::RPCEnv() {
-#ifndef _WIN32
+#if defined(ANDROID) || defined(__ANDROID__)
+  char cwd[PATH_MAX];
+  auto cmdline = fopen("/proc/self/cmdline", "r");
+  fread(cwd, 1, sizeof(cwd), cmdline);
+  fclose(cmdline);
+  base_ = "/data/data/" + std::string(cwd) + "/cache/rpc";
+#elif !defined(_WIN32)
```

Comment:
This change has the effect of now requiring a user to provide a target path to `RPCSession.upload(target=path)`, when running from `/data/local/tmp` for example. Without providing the target path, the `/data/data` prefix is always prepended resulting in errors like: 
```
   Check failed: !fs.fail() == false: Cannot open /data/data//data/local/tmp/tvm_rpc/cache/rpc/dev_lib_cl.so
```

Reply:
@csullivan great catch, can you send a new PR to fix this problem?

Reply:
Absolutely, please allow me some time.